app/mender.go

```diff
@@ -483,6 +511,11 @@ func transitionState(to State, ctx *StateContext, c Controller) (State, bool) {
 		}
 	}
 
+	spin := true
```

Comment:
sort of horrid code, looking at it now, this might as well be in the `spinEventLoop` function...

Reply:
Yes, and it looks like you've already done it, so this can be removed. I agree it's better to spin inside the function instead of here.